services/sgx/src/lib.rs

```diff
@@ -154,6 +155,27 @@ pub fn main() {
     enclave_builder.usercall_extension(ExternalService { attest_state });
     let enclave = enclave_builder.build(&mut device).unwrap();
 
+    std::thread::spawn(|| {
+        let state_pub_key = block_on(async move { fn_sdk::api::fetch_sgx_shared_pub_key().await });
+
+        // TODO(matthias): until we update the enclave to send the public key along with the
+        // encrypted shared secret key, we have to hit the http endpoint in the enclave to get the
+        // public key.
+        // Wait for enclave to start up.
+        std::thread::sleep(Duration::from_secs(10));
```

Comment:
Would it make sense to poll `http://127.0.0.1:8011/key` until it's available instead of sleeping here?

Reply:
Yeah that makes sense. I didn't go for it because there is a small bug in the enclave http server. Sometimes the requests will hang. If this happens, then the wait loop is stuck. I guess we can put a timeout on the wait loop. All of this will be fixed in the next version of the enclave, but for the next version of the enclave we don't even have to do this anymore, because we will just return the public key from the enclave.